deepspeed/runtime/zero/stage3.py

```diff
@@ -2115,6 +2153,26 @@ def get_fp32_grad_partitions(self) -> Dict[int, Dict[int, Tensor]]:
 
         return grad_dict
 
+    def get_fp32_grad_for_param(self, param) -> Tensor:
+        self.__reduce_and_partition_stream.synchronize()
```

Comment:
```suggestion
    def get_fp32_grad_for_param(self, param) -> Tensor:

        if not param.requires_grad:
            return None

        self.__reduce_and_partition_stream.synchronize()
```
this takes care of frozen params